core/pva/src/main/java/org/epics/pva/client/CreateChannelHandler.java

```diff
@@ -52,7 +52,9 @@ public void handleCommand(final ClientTCPHandler tcp, final ByteBuffer buffer) t
 
             // Reset channel to init state and search again, after delay
             channel.setState(ClientChannelState.INIT);
-            tcp.getClient().search.register(channel, false);
+            try (final var client = tcp.getClient()) {
```

Comment:
`tcp.getClient()` returns the client that we continue to use.
It must not be closed just because we are adding one more channel to it, quite the contrary, we add the channel because we fully intent to use it.